src/metabase/query_processor/middleware/annotate.clj

```diff
@@ -159,6 +159,9 @@
            (select-keys (infer-expression-type expression) type-info-columns)))
        clauses))
 
+    (mbql.u/is-clause? #{:datetime-add :datetime-subtract} expression)
+    (select-keys (infer-expression-type (second expression)) type-info-columns)
```

Comment:
Discovered an issue with this.
Adding a part that is less than a day will return the wrong :base_type. 

e.g. this test fails
```
  (is (= {:base_type :type/DateTime
          :semantic_type nil}
         (infered-col-type [:datetime-add $date 2 :hour] checkins)))
```

Reply:
Resolving, I confused the semantics of datetime-add, same as this comment: https://github.com/metabase/metabase/pull/26279#discussion_r1017146107